### PR TITLE
TBS Add generic timeout message for Progressive Enrollment

### PR DESCRIPTION
**Pull request by brentschaus-okta**
## Description
- **What's changed?** Adds the new timeout error for Progressive Enrollment using Registration Inline Hooks. Tweaks the formatting of the existing content. Also, while under the hood, updates the sample requests/response in the ref doc (based on what's already published in the registration inline hook guide).
- **Is this PR related to a Monolith release?** Yes. 2022.05.3

### Netlify link:
https://6286edc0f1c2e26d8beb8ce9--reverent-murdock-829d24.netlify.app/docs/reference/registration-hook/#timeout-behavior

### Resolves:
* [OKTA-499896](https://oktainc.atlassian.net/browse/OKTA-499896)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md [hunk 1]
```diff
@@ -198,141 +198,216 @@ Registrations are allowed by default, so setting a value of `ALLOW` for the `act
...
-For the Registration Inline Hook, the `error` object provides a way of displaying an error message to the end user who is trying to register.
+For the Registration Inline Hook, the `error` object provides a way of displaying an error message to the end user who is trying to register or update their profile.
+
+> **Note:** Generic Progressive Enrollment error messages are only available in the Identity Engine.
 
 * If you're using the Okta Sign-In Widget for Profile Enrollment, only the `errorSummary` messages of the `errorCauses` objects that your external service returns appear as inline errors, given the following:
 
    * You don't customize the error handling behavior of the widget.
-   * The `location` of `errorSummary` in the `errorCauses` object specifies the request object's user profile attribute. See [JSON response payload objects - error](/docs/concepts/inline-hooks/#error).
+   * The `location` of `errorSummary` in the `errorCauses` object specifies the request object's user profile attribute.
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:208]: I question this bullet point. Namely, even if it's required. I'd need to do some testing, but leaving this attribute blank, I believe, will not have an affect on where the UI message appears. Or, can error messages be tied, inline, to the appropriate progressive profiling field? The `location` field is added from the developer as a string field to aid in troubleshooting, according to our doc. https://developer.okta.com/docs/concepts/inline-hooks/#error

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md [hunk 2]
```diff
@@ -91,11 +91,11 @@ The action is `ALLOW` by default (in practice, `DENY` will never be sent to your
 
 Using the `com.okta.action.update` [command](#supported-commands) in your response, you can change the action that Okta will take.
 
-### data.context.user
-
-<ApiLifecycle access="ie" /><br>
+<!-- Need to clarify if we need to include this in the docs. right now, it has no content. in the context of the registration inline hook guide, it allows customers to include the object in the console.log().
```
[the remark this answers is not in the document]
Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:94]: Okay, I'll leave it commented out for now. Thanks!

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md (hunk 1 above)
Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:208]: @ericalong-okta We worked on this together. What are your thoughts?

Comment (ericalong-okta) [reply packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:208]: @brianduffield-okta it'll affect the UI error message if the `location` attribute is left blank, the improvement was requested by [OKTA-473152](https://oktainc.atlassian.net/browse/OKTA-473152). The error messages are now tied and inline to SIW fields for both registration (v1), SSR and PE (oie).

**Review by ericalong-okta — APPROVED** [review]
lgtm!

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md [hunk 3]
```diff
@@ -19,18 +19,18 @@ For a general introduction to Okta Inline Hooks, see [Inline Hooks](/docs/concep
...
 For an example implementation of this Inline Hook, see [Registration Inline Hook](/docs/guides/registration-inline-hook/nodejs/main/).
 
 ## About
 
-The Okta Registration Inline Hook allows you to integrate your own custom code into Okta's [Profile Enrollment](https://help.okta.com/okta_help.htm?type=oie&id=ext-create-profile-enrollment) flow. The hook is triggered after Okta receives the registration request but before the user is created. Your custom code can:
+The Okta Registration Inline Hook allows you to integrate your own custom code into Okta's [Profile Enrollment](https://help.okta.com/okta_help.htm?type=oie&id=ext-create-profile-enrollment) flow. The hook is triggered after Okta receives the registration or profile update request. Your custom code can:
 
-- Set or override the values that will be populated in attributes of the user's Okta profile
 - Allow or deny the registration attempt, based on your own validation of the information the user has submitted
+- Set or override the values that will be populated in attributes of the user's Okta profile
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:31]:
```suggestion
- Set or override the values that are populated in attributes of the user's Okta profile
```

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md (hunk 2 above)
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:94]:
```suggestion
<!-- Need to clarify if we need to include this in the docs. right now, it has no content. in the context of the registration inline hook guide, it allows customers to include the object in the console.log().-->
```

**Review by susanharper-okta — APPROVED** [review]
LGTM, two nits - approving to not hold you up.

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md [hunk 4]
```diff
@@ -91,11 +91,11 @@ The action is `ALLOW` by default (in practice, `DENY` will never be sent to your
...
 Using the `com.okta.action.update` [command](#supported-commands) in your response, you can change the action that Okta will take.
 
-### data.context.user
-
-<ApiLifecycle access="ie" /><br>
+<!-- Need to clarify if we need to include this in the docs. right now, it has no content. in the context of the registration inline hook guide, it allows customers to include the object in the console.log().
 
+### data.context.user
 
+<ApiLifecycle access="ie" /><br> -->
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:98]: Hi @susanharper-okta I put the closing comment "tag" here. Without it, we have an orphan title with no content :)

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md (hunk 2 above)
Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:94]: See comment on line 98.

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md (hunk 4 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:98]: gotcha. now I see it. thanks!

**Review by susanharper-okta — APPROVED** [review]
LGTM

**Pull request merged**